javatests/arcs/core/storage/LocalStorageEndpointManagerTest.kt

```diff
@@ -0,0 +1,121 @@
+package arcs.core.storage
+
+import arcs.core.crdt.CrdtData
+import arcs.core.crdt.CrdtOperationAtTime
+import arcs.core.data.EntityType
+import arcs.core.data.Schema
+import arcs.core.data.SchemaFields
+import arcs.core.data.SchemaName
+import arcs.core.data.SingletonType
+import arcs.core.storage.keys.RamDiskStorageKey
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.core.storage.testutil.testStorageEndpointManager
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import kotlinx.coroutines.withTimeout
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(JUnit4::class)
+class LocalStorageEndpointManagerTest {
+
+  private val emptyCallback: ProxyCallback<CrdtData, CrdtOperationAtTime, Any> = {}
+
+  private val storageOptions = StoreOptions(
+    storageKey = ReferenceModeStorageKey(
+      RamDiskStorageKey("backing"),
+      RamDiskStorageKey("entity")
+    ),
+    type = SingletonType(
+      EntityType(
+        Schema(
+          setOf(SchemaName("TestType")),
+          fields = SchemaFields(
+            emptyMap(), emptyMap()
+          ),
+          hash = "abcdef"
+        )
+      )
+    )
+  )
+
+  private fun runTest(block: suspend CoroutineScope.() -> Unit): Unit = runBlockingTest {
+    block()
+  }
+
+  @Test
+  fun manager_get_createsNewStore() = runTest {
+    val endpointManager = testStorageEndpointManager(this)
+
+    val firstEndpoint = withTimeout(15000) {
+      endpointManager.get(storageOptions, emptyCallback)
+    }
+
+    val secondEndpoint = withTimeout(15000) {
+      endpointManager.get(
```

Comment:
To help with test readability, it might be useful to make a helper method like:

```
fun dummyOptionsForKey(storageKey: StorageKey)
```

And to declare `DUMMY_STORAGE_KEY_1` and `DUMMY_STORAGEKEY_2` below.

Then in the test, you can use

`endpointManager.get(storeOptionsForKey(DUMMY_STORAGEKEY_1))`

This helps make it very clear in the test exactly which preconditions are being tested.